File: Python/aprendizaje/Ejercicios/Alumnos.py
"""
10. Crear un script que reciba la nota de 15 alumnos. Mostrar cuantos han aprobado y cuantos han suspendido.

"""

aprobados = 0
reprobados = 0

for i in range(1,16):
    while True:
        nota = int(input(f"Ingrese la nota del alumno {i}: "))
        if nota >= 0 and nota <= 10:
            break
    if nota >= 6:
        aprobados += 1
    else:
        reprobados += 1

print(f"Hay {aprobados} aprobados y {reprobados} reprobados")

File: Rust/basicos/Alumnos.rs
// 10. Crear un script que reciba la nota de 15 alumnos. Mostrar cuantos han aprobado y cuantos han suspendido.

use std::io;

fn input() -> i32 {
    loop {
        let mut input = String::new();
        let _ = io::stdin().read_line(&mut input);
        let n = input.trim();
        match n.parse::<i32>(){
            Ok(n) => {
                if n >= 0 && n <= 10 {
                    break n;
                }else {
                    println!("Ingrese una calificación dentro del rango: ");
                }
            }
            Err(e) => println!("Error: {}", e)
        }
    }
}

fn main() {
    let mut aprobados:i32 = 0;
    let mut reprobados:i32 = 0;

    for i in 1..=15 {
        println!("Ingrese la calificación del alumno {}: ", i);
        let calificacion:i32 = input();
        if calificacion >= 6 {
            aprobados += 1;
        }else{
            reprobados += 1;
        }
    }
    println!("Aprobados: {}", aprobados);
    println!("Reprobados: {}", reprobados);
}
